src/test/app/Flow_test.cpp

```diff
@@ -1254,6 +1254,34 @@ struct Flow_test : public beast::unit_test::suite
             ter(temBAD_PATH));
     }
 
+    template<class... T>
+    void testWithFeats(bool haveFlow, T&&... fs)
+    {
+        testLineQuality({ fs... });
+        testFalseDry({ fs... });
+        if (!haveFlow)
+            return;
+        testDirectStep({ fs... });
+        testBookStep({ fs... });
+        testDirectStep({ featureOwnerPaysFee, fs... });
+        testBookStep({ featureOwnerPaysFee, fs... });
+        testTransferRate({ featureOwnerPaysFee, fs... });
+        testSelfPayment1({ fs... });
+        testSelfPayment2({ fs... });
+        testSelfFundedXRPEndpoint(false, { fs... });
+        testSelfFundedXRPEndpoint(true, { fs... });
+        testUnfundedOffer(true, { fs... });
+        testUnfundedOffer(false, { fs... });
+        testReexecuteDirectStep({ fix1368, fs... });
+        testSelfPayLowQualityOffer({ fs... });
+    }
+
+    template<class... T>
+    void testWithFeats(T&&... fs)
+    {
+        testWithFeats(!!sizeof...(fs), fs...);
```

Comment:
I prefer `sizeof...(fs) > 0` or `sizeof...(fs) != 0` to `!!`. But that's clearly just a personal preference, no change needed. Plus this code will go away later anyway.

Reply:
I'll leave it for later, then.